autoload/airline/extensions/branch.vim

```diff
@@ -109,6 +110,49 @@ function! s:get_hg_branch()
   return ''
 endfunction
 
+if s:has_async
+  let s:jobs = {}
+
+  function! s:on_stdout(channel, msg) dict abort
+    let self.buf .= a:msg
+  endfunction
+
+  function! s:on_exit(channel) dict abort
+    let untracked = get(g:, 'airline#extensions#branch#notexists', g:airline_symbols.notexists)
+    if empty(self.buf)
+      let s:untracked_{self.cmd}[self.file] = ''
+    elseif (self.buf[0:1] is# '??' && self.cmd is# 'git') || (self.buf[0] is# '?' && self.cmd is# 'hg')
+      let s:untracked_{self.cmd}[self.file] = untracked
+    else
+      let s:untracked_{self.cmd}[self.file] = ''
+    endif
+  endfunction
+
+  function! s:DoAsync(cmd, file)
```

Comment:
Consider adding a more descriptive name, like `get_vcs_untracked_async`. `cmd` argument makes it seem to be a general function, but `s:on_exit` specifically updates `untracked`.